portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACSettings.macro

```diff
@@ -40,6 +40,35 @@ definition {
 			value1 = "${pageViewsUsageValue}");
 	}
 
+	macro assertPlanDetails {
+		var key_planName = "${planName}";
+		AssertElementPresent(locator1 = "ACUsage#PLAN_NAME");
+
+		for (var limits : list "Individuals-${individualsLimit},Page Views-${pageViews},Users-${usersLimit}") {
+			var planItem = StringUtil.extractFirst("${limits}","-");
+			var planLimits = StringUtil.extractLast("${limits}","-");
+			var tableName = "limits";
+
+			AssertElementPresent(
+				key_planItem = "${planItem}",
+				key_planLimits = "${planLimits}",
+				key_tableName = "${tableName}",
+				locator1 = "ACUsage#USAGE_PLAN_LIMITS_TABLE");
+		}
```

Comment:
Hi @AfonsoHSantos , what do you think about improving the code legibility by removing the ``for`` loops and the String manipulation... I thought in something like this:
```

AssertElementPresent(
				key_planItem = "Individuals",
				key_planLimits = "${individualsLimit}",
				key_tableName = "limits",
				locator1 = "ACUsage#USAGE_PLAN_LIMITS_TABLE");

AssertElementPresent(
				key_planItem = "Page Views",
				key_planLimits = "${pageViews}",
				key_tableName = "limits",
				locator1 = "ACUsage#USAGE_PLAN_LIMITS_TABLE");

AssertElementPresent(
				key_planItem = "Users",
				key_planLimits = "${usersLimit}",
				key_tableName = "limits",
				locator1 = "ACUsage#USAGE_PLAN_LIMITS_TABLE");
```

And the same breakdown logic for the addon's section.

Reply:
+1

Reply:
+1

Reply:
fixed